### PR TITLE
implement description of a message and description of a transaction

### LINKED ISSUE #189: Implement function to get description of a message / description of a transaction

### PR DESCRIPTION
**Pull request by prin-r**
Fixes #189

### CODE REVIEW

File: scan/src/rpc/TxHook.re [hunk 1]
```diff
@@ -108,10 +112,42 @@ module Msg = {
...
+      send.amount
+      ->Belt_List.map(coin => coin->Coin.getDescription)
+      ->Belt_List.reduceWithIndex("", (des, acc, i) =>
+          acc
+          ++ des
+          ++ {
+            i + 1 < send.amount->Belt_List.size ? "," : "";
+          }
+        )
+      ++ "→"
```
Comment (sorawit) [inline scan/src/rpc/TxHook.re]: Does this actually work? did you verify?

File: scan/src/rpc/TxHook.re [hunk 2]
```diff
@@ -108,10 +112,42 @@ module Msg = {
...
+      ++ (send.toAddress |> Address.toBech32)
+    | Store(store) => store.name
+    | Request(_) => _getCodeName(msg, "request")
```
Comment (sorawit) [inline scan/src/rpc/TxHook.re]: From the design it should have request number too, right?

File: scan/src/rpc/TxHook.re [hunk 3]
```diff
@@ -108,10 +112,42 @@ module Msg = {
...
+  let getDescription = msg => {
+    switch (msg.action) {
+    | Send(send) =>
+      send.amount
+      ->Belt_List.map(coin => coin->Coin.getDescription)
+      ->Belt_List.reduceWithIndex("", (des, acc, i) =>
+          acc
+          ++ des
+          ++ {
+            i + 1 < send.amount->Belt_List.size ? "," : "";
```
Comment (sorawit) [inline scan/src/rpc/TxHook.re]: Use paren, remove semicolon. Also we should add space after comma?

File: scan/src/rpc/TxHook.re [hunk 4]
```diff
@@ -108,10 +112,42 @@ module Msg = {
...
+    switch (msg.action) {
+    | Send(send) => send.fromAddress
+    | Store(store) => store.owner
+    | Request(request) => request.sender
+    | Report(report) => report.validator
+    | Unknown => "" |> Address.fromHex
+    };
+  };
+
+  let _getCodeName = (msg, prefix) =>
```
Comment (sorawit) [inline scan/src/rpc/TxHook.re]: Let's write a generic function in `Event` to get a value under a specified key.

**Review by sorawit — CHANGES_REQUESTED** [review]

File: scan/src/components/TxIndexPageTable.re [hunk 5]
```diff
@@ -52,8 +52,8 @@ let make = (~messages: list(TxHook.Msg.t)) => {
       </Row>
     </THead>
     {messages
-     ->Belt.List.map(msg => {
-         <TBody>
+     ->Belt.List.mapWithIndex((index, msg) => {
+         <TBody key={index |> string_of_int}>
```
Comment (sorawit) [inline scan/src/components/TxIndexPageTable.re:56]: Do not use index here.

File: scan/src/rpc/TxHook.re [hunk 6]
```diff
@@ -20,6 +20,17 @@ module Event = {
 
   let decodeEvents = json =>
     List.flatten(JsonUtils.Decode.(json |> field("events", list(decodeEvent))));
+
+  let getValueOfKey = (events: list(t), key) => {
+    let hasEvent = events->Belt_List.some(event => event.key == key);
```
Comment (sorawit) [inline scan/src/rpc/TxHook.re]: Let's do this. 
`events->Belt.List.keepMap((event) => event.key == key ? Some(event.value) : None)->Belt.List.get(0)`

**Review by sorawit — CHANGES_REQUESTED** [review]

File: scan/src/reusable/TElement.re [hunk 7]
```diff
@@ -1,7 +1,12 @@
 module Styles = {
   open Css;
 
-  let typeContainer = style([marginRight(`px(20)), maxWidth(`px(210))]);
+  let typeContainer = w =>
+    style([
+      marginRight(`px(20)),
+      maxWidth(`px(210)),
```
Comment (evilpeach) [inline scan/src/reusable/TElement.re]: remove `maxWidth`

**Review by evilpeach — APPROVED** [review]
LGTM

**Review by sorawit — APPROVED** [review]
lgtm

**Pull request merged**